esphome/components/host/__init__.py

```diff
@@ -34,6 +34,11 @@ def set_core_data(config):
 
 
 async def to_code(config):
-    cg.add_build_flag("-DUSE_HOST")
+    cg.add_build_flag("-DUSE_HOST=1")
```

Comment:
This should stay with no value but be checked with `ifdef` as per all of our other platform checks
```suggestion
    cg.add_build_flag("-DUSE_HOST")
```